### DISCUSSION TITLE
Is it necessary to wait after startServer()?

### DISCUSSION THREAD

**Discussion by AlphonsusNguyen** [moquette-io/moquette#783] [Q&A] [answered]
Hi everyone :) , 
    I have a question about the Server.startServer() method. After calling this method, do we need to wait a bit for the broker server to finish "starting up" and be ready to process the messages? If yes, is there a listener that can notify the app when the server is fully operational?

Thanks in advance.

**Comment by andsel** [comment, marked as answer]
Hi @AlphonsusNguyen actually there isn't yet an asynchronous way to get notified when the server is up and ready.
I'll provide it in the future, but as today you can sleep (the startup it's really fast) or you could ping the MQTT interface with a polling client that tries to connect.

**Reply by AlphonsusNguyen** [reply]
Thank you.
